docs/porposals/diamond.md

```diff
@@ -0,0 +1,40 @@
+# Diamond workflow proposal.
+
+Diamond workflow allows for a pipeline to scale out the generation of a single bundle. There 
+is little or no co-ordination between the nodes. Each node uploads part of the bundle in the
+ filesystem namespace of the bundle. 
+
+The diamond workflow consists of the following steps.
+
+1. Before a diamond starts, Datamon SDK allows for the creation of an unique ID for a diamond.
+   1. This can be a KSUID.
+   2. The client is expected to use the KSUID at each step.
+   3. command: ```datamon bundle split initialize --repo <repo> --message```
+   4. backend model: In the vmetadata bucket the following path will be created.
+    ```/splits/repo/<id>/split.yaml```
+   
+2. Each node in a diamond generates a filelist that is stored in the versioned metadata folder.
+   1. Each filelist is given a client assigned name. Example node1,node2,node3 or 1,2,3 or a UUID.
+   2. On retries the name should be the same for a given node.
+      1. A node on failure can rerun the same command, as long as the split-name is the same
+      the previous filelist contents will be superseded by the ones uploaded later.
+   3. command: ```datamon bundle split add --repo <repo> --split-id <id> --source <path>```
```

Comment:
How is the node name being specified by the client in this command? Shouldn't there be a `--node-name` argument?

Reply:
```split-id``` is the same as ```node-name``` will clean up nomenclature in the document.